spec/helpers/application_helper_spec.rb

```diff
@@ -41,4 +41,78 @@
       expect(helper.library_name('NOSUCHLIBRARY')).to eq 'NOSUCHLIBRARY'
     end
   end
+
+  describe '#render_checkout_status' do
+    let(:checkout) do
+      instance_double(
+        Checkout,
+        recalled?: false,
+        overdue?: false,
+        lost?: false,
+        claimed_returned?: false,
+        accrued: 0
+      )
+    end
+    let(:content) { Capybara.string(helper.render_checkout_status(checkout)) }
+
+    context 'when a recalled item has accrued fines' do
+      before do
+        allow(checkout).to receive_messages(recalled?: true, accrued: 15)
```

Comment:
Is there a reason we're using `allow` here instead of `expect`?

Reply:
I tried setting this `allow` to an `expect` within an `it` block instead of a `before` block, and got a couple of spec failures:

```
 1) ApplicationHelper#render_checkout_status when a recalled item has accrued fines renders the right html
     Failure/Error: expect(content).to have_css('.text-recalled', text: 'Recalled $15').and(have_css('.sul-icons'))
     
     NoMethodError:
       undefined method `css' for nil:NilClass
```

and

```
2) ApplicationHelper#render_checkout_status when a recalled item has accrued fines is recalled with accrued fines
     Failure/Error: expect(checkout).to receive_messages(recalled?: true, accrued: 15)
     
       (InstanceDouble(Checkout) (anonymous)).recalled?(no args)
           expected: 1 time with any arguments
           received: 0 times
     # ./spec/helpers/application_helper_spec.rb:60:in `block (4 levels) in <top (required)>'
```

so it looks like in this case the test object can't receive methods the next expectation requires...

Reply:
Hmm, okay, it seems like if we either allowed ourselves to use `expect` in `before` or multiple `expect`s in a single test we could get around this, but it's not a big deal I suppose.